pkg/distributor/push.go

```diff
@@ -190,6 +190,30 @@ func handler(
 	})
 }
 
+// Retryable HTTP codes for otlp collector
+var retryableHTTPCodes = map[int]bool{
+	502: true,
+	503: true,
+	504: true,
+	429: true,
+}
+
+// convertHTTPCodeOtlp maps specific HTTP status codes to those recognized by the OTLP client for retry.
+//   - Maps a custom 529 error to the standard 429 (Too Many Requests) error.
+//   - Converts non-retryable 5xx errors to 503 (Service Unavailable) to trigger client retries, as per OTLP specifications (https://opentelemetry.io/docs/specs/otlp/#failures-1).
+//   - Ensures that the 501 (Not Implemented) error is not retried.
+var convertHTTPCodeOtlp = func(httpCode int) int {
+	if httpCode == StatusServiceOverloaded {
+		httpCode = http.StatusTooManyRequests
+	}
+	if httpCode/100 == 5 && !retryableHTTPCodes[httpCode] {
```

Comment:
Stylistically, I would rather do this as a switch than a map; that way the compiler has the best chance to produce efficient code.

Reply:
Good point. I think it also doesn't hurt to avoid a global variable (currently it's named generically, and not tied to OTLP).

Reply:
it is removed, since we decided to dublicated code instead.